packages/kit/src/runtime/client/fetcher.js

```diff
@@ -112,7 +130,17 @@ export function subsequent_fetch(resource, resolved, opts) {
 		}
 	}
 
-	return native_fetch(resolved, opts);
+	const patched_opts = { ...opts };
+	if (DEV) {
+		// This assigns the __sveltekit_fetch__ flag and makes it non-enumerable
+		Object.defineProperty(patched_opts, '__sveltekit_fetch__', {
+			value: true,
+			writable: true,
+			configurable: true
+		});
```

Comment:
It's `DEV` only where bundle size isn't a concern, so that's fine. I suggest to do the `patched_opts` in `DEV` mode only, too - basically something like
```js
if (DEV) {
  const patched_opts = { ...opts };
  //..
  return window.fetch(resolved, patched_opts);
} else {
  return window.fetch(resolved, opts);
}
```
(similar at the other call site)
.. unless there's a reason we need a shallow copy of the opts?

Reply:
I don't think we need to shallow copy the `opts`. If someone's patched `fetch`, that's on them to make sure that that version doesn't unexpectedly mutate its arguments.